### DISCUSSION TITLE
Problem about inference with C++ interface

### DISCUSSION THREAD

**Discussion by KawaSakiHakuButsu** [deepmodeling/deepmd-kit#1493] [Q&A] [answered]
Hello everyone.

I have __deepmd-kit 2.0.3-gpu__ installed with __*install.sh__. And I try to run [the example of inference with C++ interface](https://github.com/deepmodeling/deepmd-kit/blob/master/doc/inference/cxx.md), but failed with compile error.

<img width="767" alt="屏幕截图 2022-02-22 163636" src="https://user-images.githubusercontent.com/82147508/155093977-df5e0fb8-63a3-4ea8-9d2b-8a6bd65b9674.png">
(In my try, I set "deepmd_root=/home/JinB/deepmd-kit/2.0.3, tensorflow_root=/home/JinB/deepmd-kit/2.0.3")

So do I have to install extra tensorflow C++ version, or is there an easier solution for the problem?
Besides, how can I use the C++ interface through cmake?

Thanks you very much.

**Comment by njzjz** [comment, marked as answer]
<del>Looks like you didn't use the consistent compiler. Especially, do not use GCC 4.</del>

**Reply by KawaSakiHakuButsu** [reply]
I have tried various __gcc__ of version 7.2.0, 7.5.0, 8.3.0, 9.3.0, 11.1.0, but ended up with the same wrong message.

**Reply by njzjz** [reply]
I can reproduce the error. `-ltensorflow_cc` should be added here.

**Reply by njzjz** [reply]
Fixed in #1504.

**Reply by KawaSakiHakuButsu** [reply]
I have solved the problem, according to your suggestions [#1504.](https://github.com/deepmodeling/deepmd-kit/pull/1504)
Thank you very much.
